src/iot/commands/createPolicy.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as fs from 'fs-extra'
+import { getLogger } from '../../shared/logger'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { IotPolicyFolderNode } from '../explorer/iotPolicyFolderNode'
+
+/**
+ * Creates a policy from a policy document.
+ */
+export async function createPolicyCommand(
+    node: IotPolicyFolderNode,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('CreatePolicy called for %O', node)
+
+    const policyName = await window.showInputBox({
+        prompt: localize('AWS.iot.createPolicy.prompt', 'Enter a new policy name'),
+        placeHolder: localize('AWS.iot.createPolicy.placeHolder', 'Policy Name'),
+    })
+
+    if (!policyName) {
+        getLogger().info('CreatePolicy canceled')
+        return
+    }
+
+    const fileLocation = await promptForPolicyLocation(window)
+    if (!fileLocation) {
+        getLogger().info('CreatePolicy canceled: No document selected')
+        return
+    }
+
+    try {
+        const data = await fs.readFile(fileLocation.fsPath)
+        //Parse to ensure this is a valid JSON
+        const policyJSON = JSON.parse(data.toString())
+        await node.iot.createPolicy({ policyName, policyDocument: JSON.stringify(policyJSON) })
+    } catch (e) {
+        getLogger().error('Failed to create policy document: %O', e)
+        showViewLogsMessage(localize('AWS.iot.createPolicy.error', 'Failed to create policy'), window)
+        return
+    }
+
+    //Refresh the Policy Folder node
+    await node.refreshNode(commands)
+}
+
+export async function promptForPolicyLocation(window: Window): Promise<vscode.Uri | undefined> {
```

Comment:
I exported this function from this file, since it is also used in `createPolicyVersion.ts`. Is that acceptable, or should it be moved to its own utility file?

Reply:
If it's just one shared function I think it's ok here.